enhancements/insights/config-api.md

```diff
@@ -0,0 +1,296 @@
+	SimpleContentAccess SimpleContentAccess `json:"simpleContentAccess,omitempty"`
+	// ClusterTransfer spec attribute includes all the configuration options related to
+	// periodical check of pending cluster transfers.
+	ClusterTransfer ClusterTransfer `json:"clusterTransfer,omitempty"`
+}
+
+type InsightsStatus struct {
+	// GatheringStatus provides basic information about the last Insights gathering.
+	GatheringStatus GatheringStatus `json:"gatheringStatus,omitempty"`
+	// SimpleContentAccessStatus provides basic information about the SCA controller.
+	SimpleContentAccessStatus SimpleContentAccessStatus `json:"simpleContentAccessStatus,omitempty"`
+	// ClusterTransferStatus provides basic information about the ClusterTransfer controller.
+	ClusterTransferStatus ClusterTransferStatus `json:"clusterTransferStatus,omitempty"`
+	// ReportStatus provides general Insights analysis results.
+	ReportStatus ReportStatus `json:"insightsReport,omitempty"`
+}
+
+type Gathering struct {
+	// Interval defines the time frequency of gathering. The duration is 2h by default.
+	// +kubebuilder:default=2h
+	Interval time.Duration `json:"interval,omitempty"`
+	// UploadEndpoint defines ingress endpoint for uploading Insights archive
+	// +kubebuilder:default=https://cloud.redhat.com/api/ingress/v1/upload
+	UploadEndpoint string `json:"endpoint,omitempty"`
+	// EnableObfuscation is flag enabling the global obfuscation of the IP addresses and base domain
+	// in the Insights archive data.
+	EnableObfuscation bool `json:"enableObfuscation,omitempty"`
+	// Enables user to force Insights data gathering by setting a new reason
+	ForceGatheringReason string `json:"forceGatheringReason,omitempty"`
+	// List of gatherers to be included in the gathering. Empty list means that all gatherers are included
+	Gatherers []string `json:"gatherers,omitempty"`
+	// InsightsReport includes all the configuration options related to downloading the Insights report.
+	InsightsReport InsightsReport `json:"insightsReport,omitempty"`
+	// Flag to disable gathering. When the gathering is disabled then it cannot be forced with the `ForceGatheringReason`
+	Disabled bool `json:"disabled,omitempty"`
+	// Path in the container for temporary storage of the Insights archive
+	// +kubebuilder:default=/var/lib/insights-operator
+	StoragePath string `json:"storagePath,omitempty"`
+}
+
+type SimpleContentAccess struct {
+	// Endpoint is the HTTP endpoint to pull down the SCA certificates from. This is set to
+	// corresponding OCM API by default.
+	// +kubebuilder:default=https://api.openshift.com/api/accounts_mgmt/v1/certificates
+	Endpoint string `json:"endpoint,omitempty"`
+	// Interval defines the time frequency of downloading the SCA certificates.
+	// The duration is 8h by default.
+	// +kubebuilder:default=8h
+	Interval time.Duration `json:"interval,omitempty"`
+	// Disabled flag is an option how to disable the SCA download feature.
+	Disabled bool `json:"disabled,omitempty"`
```

Comment:
Yes, it's an option. This would require mentioning the endpoint value in the documentation, because if you want to enable it back, you must find the original value.

Reply:
Is an extra bool attribute the problem? I think I would prefer to have an explicit disable option here. My point is basically the above - I don't want to maintain the default values in the documentation.

Reply:
Every extra field is a problem :-) There is real cost associated with every single configuration option. Minimizing them improves how much we need to support them.

Reply:
OK. I am wondering if moving all this stuff to some configmap would be an alternative then....

Reply:
Moving it to a configmap doesn't reduce complexity for users or testing permutations. And if anything makes it harder for users to understand the options that exist and specify them correctly

Reply:
I removed it from the API proposal. We can add the `unsupportedConfigOverrides` field or keep the original `support` secret way of configuring it.